### PR TITLE
feat: add configurable channel folder template for storage paths

### LINKED ISSUE #1019: Allow customization of channel folder naming in storage structure
### Expected Behavior

The storage folder structure should be resilient to channel username changes and provide a stable, predictable path for VOD storage. Ideally:

1. All VODs from the same channel should remain in a single folder, regardless of username changes
2. Third-party tools accessing the storage should be able to rely on a consistent folder structure
3. Users should have control over how the top-level channel folder is named, similar to how `folder_template` and `file_template` work for VOD subfolders

### Current Behavior

The channel folder name is hardcoded to use the Twitch username (`channel.Name`):

```
{VIDEOS_DIR}/{channel_name}/{folder_template}/files...
```

This causes issues when a Twitch channel changes their username:
- A new folder is created with the new username
- Existing VODs remain in the old username folder
- VODs from the same channel end up split across multiple folders
- The database paths still point to the old location, but new archives go to the new location

Related issue: #1018

Additionally, the current structure is not ideal for third-party tools that may want to access the storage in a more structured/programmatic way, as usernames are mutable and not guaranteed to be stable identifiers.

### Possible Solution

**Option A: Use Twitch User ID as default**

Replace the channel username with the immutable `ext_id` (Twitch User ID) for the folder name:

```
{VIDEOS_DIR}/{ext_id}/{folder_template}/files...
```

This would be a breaking change for existing installations but provides a stable, permanent identifier.

**Option B: Add channel folder to the template system**

Extend the existing storage template system to include a `channel_folder_template` option with variables like:
- `{{channel}}` - Username/login
- `{{channel_id}}` - External platform ID (Twitch User ID)
- `{{channel_display_name}}` - Display name

**Option C: Hybrid approach**

Keep current behavior as default but add a global setting to switch to ID-based folders, with a migration task to rename existing folders.

### PR DESCRIPTION
**Pull request by fllppi**
## Summary
- Adds a new `channel_folder_template` config option that controls how channel-level directories are named on disk, resolving the problem of archives scattering across folders when a Twitch channel renames.
- Available template variables: `{{channel}}` (login name), `{{channel_id}}` (Twitch User ID — stable, never changes), `{{channel_display_name}}`.
- Default is `{{channel}}` which preserves existing behavior — no breaking changes.

## Details

**Problem:** Channel folders are hardcoded to use the Twitch login name. When a streamer changes their username, new archives go to a new folder while old ones remain in the old one.

**Solution:** A `channel_folder_template` setting (alongside the existing `folder_template` and `file_template`) lets users choose what names their channel directories. Using `{{channel_id}}` gives a stable identifier that never changes across renames.

**What's included:**
- New `channel_folder_template` config field with `{{channel}}` default
- All archive paths (videos, clips, livestreams, thumbnails, info files) use the resolved template
- Storage migration task extended with a two-phase approach: Phase 1 migrates channel folders, Phase 2 migrates VOD subfolders/files — with rollback on failure
- `{{channel_id}}` and `{{channel_display_name}}` also available in existing folder/file templates
- Path sanitization on all template output to prevent directory traversal
- `VideosDirMigrate` improved to detect the videos directory from channel image paths (more robust with custom templates)
- Admin settings UI with consolidated variable reference
- Tests for all new template resolution including security edge cases

Closes #1019

### CODE REVIEW

File: internal/tasks/thumbnail.go [hunk 1]
```diff
@@ -129,7 +130,16 @@ func (w GenerateSpriteThumbnailWorker) Work(ctx context.Context, job *river.Job[
...
+	channelFolderName, err := storagetemplate.GetChannelFolderName(storagetemplate.ChannelTemplateInput{
+		ChannelName:        channel.Name,
+		ChannelID:          channel.ExtID,
+		ChannelDisplayName: channel.DisplayName,
+	})
+	if err != nil {
+		log.Warn().Err(err).Msg("error resolving channel folder template, falling back to channel login name")
+		channelFolderName = channel.Name
+	}
+	rootVideoPath := fmt.Sprintf("%s/%s/%s", env.VideosDir, channelFolderName, video.FolderName)
```
[the remark this answers is not in the document]
Comment (fllppi) [reply internal/tasks/thumbnail.go]: adressed in [2fc58c5](https://github.com/Zibbp/ganymede/pull/1085/commits/2fc58c5c11f4bbd5d811857211377e11af42e35b)

**Comment by fllppi** [comment]
@coderabbitai resume

**Comment by fllppi** [comment]
@coderabbitai review

**Comment by fllppi** [comment]
Hey @Zibbp, did you have time to review it or is there anything blocking it?
